### PR TITLE
Set ingress class name

### PR DESCRIPTION
**Pull request by QuentinBisson**
## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: helm/prometheus-meta-operator/templates/ingress-oauth.yaml [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
+            port:
+              number: 9090
+          {{- else }}
           serviceName: oauth2-proxy
           servicePort: 4180
+          {{- end }}
         path: /oauth2
+        {{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
+        pathType: ImplementationSpecific
+        {{- end }}
```
Comment (TheoBrigitte) [inline helm/prometheus-meta-operator/templates/ingress-oauth.yaml:40]: Why do we need a condition for this?

File: service/controller/resource/monitoring/ingress/resource.go [hunk 2]
```diff
@@ -114,6 +118,7 @@ func toIngress(v interface{}, config Config) (metav1.Object, error) {
 		},
 		ObjectMeta: objectMeta,
 		Spec: networkingv1beta1.IngressSpec{
+			IngressClassName: &ingressClassName,
```
Comment (TheoBrigitte) [inline service/controller/resource/monitoring/ingress/resource.go:121]: How does this works on 1.18 k8s?

Comment (QuentinBisson) [reply service/controller/resource/monitoring/ingress/resource.go:121]: I'm trying to deploy it on 1.20 first :D

Comment (QuentinBisson) [reply service/controller/resource/monitoring/ingress/resource.go:121]: Yes this is not working

Comment (QuentinBisson) [reply service/controller/resource/monitoring/ingress/resource.go:121]: I wanted to see if we could use chartutil to get the capabilities but it looks like we cannot support v1beta1 and v1 easily enough :(

Comment (TheoBrigitte) [reply service/controller/resource/monitoring/ingress/resource.go:121]: We should know the kubernetes version we are working against, but then we need to maintain both
`v1beta1.Ingress` and `v1.Ingress` objects in Go also :face_exhaling:

File: helm/prometheus-meta-operator/templates/ingress-oauth.yaml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/prometheus-meta-operator/templates/ingress-oauth.yaml:40]: I think not

File: service/controller/resource/monitoring/ingress/resource.go (hunk 2 above)
Comment (QuentinBisson) [reply service/controller/resource/monitoring/ingress/resource.go:121]: I think maintaining the 2 would not be hard tbh but that means we need to move out of the generic resource and tests :( I'm not sure I want to go that way today :D

Comment (TheoBrigitte) [reply service/controller/resource/monitoring/ingress/resource.go:121]: Maybe we can find faith together tomorrow and pair on this

Comment (QuentinBisson) [reply service/controller/resource/monitoring/ingress/resource.go:120]: Tested on gorgoth with 1.18

**Comment by QuentinBisson** [comment]
Tested on Gorgoth, gauss, ginger and gremlin

**Review by TheoBrigitte — APPROVED** [review]
Nice

**Comment by QuentinBisson** [comment]
Thanks buddy :)

**Pull request merged**